src/vscode_ocaml_platform.ml

```diff
@@ -26,6 +27,25 @@ let activate (extension : ExtensionContext.t) =
   Dune_formatter.register extension instance;
   Dune_task_provider.register extension instance;
   let sandbox = Sandbox.of_settings_or_detect () in
+  let* refresh =
+    let+ disposable, command = Treeview_switches.register extension in
+    ExtensionContext.subscribe extension ~disposable;
+    command
+  in
+  let () =
+    let handler (_ : Extension_instance.t) ~args:_ = refresh () in
+    Extension_commands.make_command Extension_consts.Commands.refresh_switches
+      handler
+    |> Extension_commands.register extension instance
+  in
+  let (_ : unit Promise.t) =
+    let+ disposable = Treeview_commands.register extension in
+    ExtensionContext.subscribe extension ~disposable
+  in
+  let (_ : unit Promise.t) =
+    let+ disposable = Treeview_help.register extension in
+    ExtensionContext.subscribe extension ~disposable
```

Comment:
More of a question: are we sure `ExtensionContext.subscribe extension ~disposable` is thread-safe? From this code, the subscriptions happens in a "background" thread, iiuc.

Reply:
I'm not sure how to test this, but:

- I did not notice any unexpected behavior that would be explained by this.
- Looking at VSCode extension codes online, subscribing in an async function is not something uncommon.